.buildbot.sh

```diff
@@ -0,0 +1,18 @@
+#! /bin/sh
+
+set -e
+
+export CARGO_HOME="`pwd`/.cargo"
+export RUSTUP_HOME="`pwd`/.rustup"
+
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh
+sh rustup.sh --default-host x86_64-unknown-linux-gnu \
+    --default-toolchain nightly \
+    --no-modify-path \
+    --profile minimal \
+    -y
+export PATH=`pwd`/.cargo/bin/:$PATH
+cargo check
+
+rustup toolchain install nightly --allow-downgrade --component rustfmt
```

Comment:
Should we also be testing this with rustgc?

Reply:
This is a bit chicken-egg at the moment. I will raise a PR for this once I've done the rustgc one, but they all need landing in a particular order. It's a bit confusing

Reply:
OK, I can buy that.